src/apps/api/views/search.py

```diff
@@ -30,11 +30,45 @@ def query(request, version="v1"):
     # Do filters
     # ...
 
+    date_flags = request.GET.get('date_flags')
+    if date_flags:
+        if date_flags == "last_month":
+            # Do something
+            s = s.filter('range', created_when={
+                'gt': datetime.date.today() - datetime.timedelta(days=30),  # I think..
+                "format": "yyyy-MM-dd",
+                'lte': datetime.date.today()
+            })
+        if date_flags == "last_year":
+            # Do something
+            s = s.filter('range', created_when={
+                'gt': datetime.date.today() - datetime.timedelta(days=365),  # I think..
+                "format": "yyyy-MM-dd",
+                'lte': datetime.date.today()
+            })
+        # ELSE DO NOTHING
+
     # Filter on dates...
-    # s = s.filter('range', created_when={
-    #     'gt': datetime.date(1988, 2, 29),
-    #     'lte': 'now'
-    # })
+    start_date = request.GET.get('start_date')
+    end_date = request.GET.get('end_date')
```

Comment:
Can do something like...

```python
date_args = {}
if start_date:
    date_args['gte'] = start_date
if end_date:
    date_args['lte'] = end_date
if date_args:
    date_args["format"] = "yyyy-MM-dd"
    s = s.filter('range', created_when=**date_args)
```

Reply:
This has been changed. I kept the cruft from the old way because I'm not sure how well this is actually working atm. Something seems off.